Add Python 3.10 to the testing

cclauss opened the pull request:
https://pythoninsider.blogspot.com/2021/10/python-3100-is-available.html

https://docs.python.org/release/3.10.0/whatsnew/changelog.html#python-3-10-0-final

The loop parameter has been removed from most of asyncio‘s high-level API following deprecation in Python 3.8.

cclauss committed:
Add Python 3.10 to the testing

https://pythoninsider.blogspot.com/2021/10/python-3100-is-available.html

https://docs.python.org/release/3.10.0/whatsnew/changelog.html#python-3-10-0-final

The loop parameter has been removed from most of asyncio‘s high-level API following deprecation in Python 3.8.

horazont commented:
@cclauss Thanks for the contribution. I expect the 3.10 pipeline to break for multiple reasons (I think my dear nosetests breaks on 3.10, too), so it's good to have it monitored.

cclauss commented:
Wha about https://github.com/nose-devs/nose2 or https://pypi.org/project/nose3 instead of nose.

horazont committed:
Remove deprecated loop arguments

Those have been deprecated since Python 3.8 (and non-mandatory since
forever), so we can just remove them for forward-compatibility.

horazont committed:
Fix use of deprecated Logger.warn

horazont committed:
Fix another deprecation for Python 3.11

(ahead of time for once!)

horazont commented:
Woop woop! Thanks a lot for triggering this, now it finally works :)
